src/components/Transcript/Transcript.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import TanscriptSelector from './TranscriptMenu/TranscriptSelector';
+import { createTimestamp, timeToMs } from '@Services/utility-helpers';
+import './Transcript.scss';
+
+const Transcript = ({ transcripts }) => {
+  const [transcript, setTranscript] = React.useState([]);
+  const [transcriptTitle, setTranscriptTitle] = React.useState('');
+  const [transcriptUrl, setTranscriptUrl] = React.useState('');
+
+  // React refs array for each timed text value in the transcript
+  let textRefs = React.useRef([]);
+  const transcriptContainerRef = React.useRef();
+
+  React.useEffect(() => {
+    if (transcripts && transcripts.length > 0) {
+      setTranscript(transcripts[0].data);
+      setTranscriptTitle(transcripts[0].title);
+      setTranscriptUrl(transcripts[0].url);
+    }
+
+    // // FIXME:: use player's current time from state management once this
+    // // is wired to it.
+    setTimeout(function () {
+      autoScrollAndHighlight(73004);
+    }, 3000);
+    setTimeout(function () {
+      autoScrollAndHighlight(27531);
+    }, 6000);
+  }, []);
+
+  const selectTranscript = (selectedTitle) => {
+    const selectedTranscript = transcripts.filter(function (tr) {
+      return tr.title === selectedTitle;
+    });
+    const { data, title, url } = selectedTranscript[0];
+    setTranscript(data);
+    setTranscriptTitle(title);
+    setTranscriptUrl(url);
+
+    setTimeout(function () {
+      autoScrollAndHighlight(73004);
+    }, 3000);
+    setTimeout(function () {
+      autoScrollAndHighlight(27531);
+    }, 6000);
+  };
+
+  const autoScrollAndHighlight = (currenttime) => {
+    let textTopOffset = 0;
+    textRefs.current.map((tr) => {
```

Comment:
`map` returns an array, so if you just looking to loop through `textRefs`, might be more clear to do a `forEach` or a `for...of` or `for...in`?